### PR TITLE
test: replace jest with vitest

### PR DESCRIPTION
**Pull request by nyapat**
**Please describe the changes this PR makes and why it should be merged:**

This is to update voice to use vitest instead of jest. It was ignored previously because the mocks are annoying to convert.
The tests compile and pass as shown by the CI.

**Status and versioning classification:**

### CODE REVIEW

**Comment by vladfrangu** [comment]
Can you try to rebase your PR? Just start the rebase, and every time you get a merge issue on pnpm-lock.yaml, just run `pnpm i`. It'll solve itself

**Review by vladfrangu — DISMISSED** [review]
Can you add a vitest config file for this too 🙏 Personally don't like relying on defaults

**Comment by vladfrangu** [comment]
Turns out only rest has a custom vitest.config.ts file 🥴. It's fine as is 🫡

File: packages/voice/package.json [hunk 1]
```diff
@@ -70,27 +70,25 @@
...
-		"@babel/core": "^7.24.6",
-		"@babel/preset-env": "^7.24.6",
-		"@babel/preset-typescript": "^7.24.6",
 		"@discordjs/api-extractor": "workspace:^",
+		"@discordjs/opus": "^0.9.0",
 		"@discordjs/scripts": "workspace:^",
-		"@favware/cliff-jumper": "^4.1.0",
-		"@types/jest": "^29.5.12",
-		"@types/node": "^16.18.105",
+		"@favware/cliff-jumper": "^3.0.3",
```
Comment (vladfrangu) [inline packages/voice/package.json:76]: revert this downgrade

File: packages/voice/package.json [hunk 2]
```diff
@@ -70,27 +70,25 @@
...
+		"@types/node": "16.18.105",
```
Comment (vladfrangu) [inline packages/voice/package.json:77]: bump to 18

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/voice/__tests__/Secretbox.test.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
+import { test, expect, vitest } from 'vitest';
 import { methods } from '../src/util/Secretbox';
 
-jest.mock('tweetnacl');
+vitest.mock('tweetnacl');
 
 test('Does not throw error with a package installed', () => {
```
Comment (vladfrangu) [inline packages/voice/__tests__/Secretbox.test.ts:6]: Well this test is wrong now, should probably adjust / delete

**Review by vladfrangu — APPROVED** [review]

File: packages/voice/vitest.config.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import defaultConfig from '../../vitest.config.js';
+
+export default defaultConfig;
```
Comment (HigherOrderLogic) [inline packages/voice/vitest.config.ts:3]: Maybe consider [Vitest workspace](https://vitest.dev/guide/workspace)? It has supports for extending config without having to import the base config into each file.

Example:
```ts
// vitest.workspace.ts at root
import { defineWorkspace } from 'vitest/config'

export default defineWorkspace([
  'packages/*',
  {
    extends: './vite.config.js',
  }
])
```

**Review by didinele — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by didinele** [comment]
Once rebased this will be automatically merged.

File: packages/voice/package.json [hunk 5]
```diff
@@ -70,27 +70,25 @@
...
+		"@types/node": "18.19.45",
+		"@vitest/coverage-v8": "2.0.5",
 		"cross-env": "^7.0.3",
 		"esbuild-plugin-version-injector": "^1.2.1",
 		"eslint": "^8.57.0",
 		"eslint-config-neon": "^0.1.62",
 		"eslint-formatter-pretty": "^6.0.1",
-		"jest": "^29.7.0",
-		"jest-websocket-mock": "^2.5.0",
 		"mock-socket": "^9.3.1",
```
Comment (almeidx) [inline packages/voice/package.json:84]: This dependency and its associated file can be removed as well
 https://github.com/nyapat/discord.js/blob/16390efe6e4f2035aa0fe365c8d593366beca7ce/packages/voice/__mocks__/ws.js

**Review by iCrawl — APPROVED** [review]

**Pull request merged**